client/src/main/java/com/paypal/selion/reports/runtime/Gatherer.java

```diff
@@ -41,7 +44,10 @@ static String saveGetLocation(WebDriver driver) {
                 location = driver.getCurrentUrl();
             }
         } catch (Exception exception) {
-            logger.log(Level.WARNING, "Current location couldn't be retrieved by getCurrentUrl().", exception);
+            logger.log(
```

Comment:
@mach6 
Here's my take on this...

We are returning a default value when we can't extract the currentUrl via the web driver. The warning is not going to do any value addition here  except for clog the console. So instead of worrying about the verbiage of the error message can we think of retaining the current error message and just change the level to `DEBUG` ? That way the console is not going to be clogged.. user is not misled and the curious ones can still find this info in the log files.

If we are determined to log this at the warning level but want to tweak the error message then I would suggest that we do something like below
```
((HasCapabilities) d).getCapabilities().getBrowserName(); // extract browser name
```
Enhance `com.paypal.selion.platform.grid.BrowserFlavors` (or create a new internal only class) to have a method which would basically accept the browser name as returned from the capability and then filter it out as `web` or `mobile` (We can have sub flavors as well) and then based on the enum it returns, we can display the corresponding error msg as well...

```
public enum ErrorType {
    APPIUM("appium", "Not supported in appium"),
    IOS("iosDriver", "Need help in implementing this in IOS Driver"),
    SELENDROID("selendroid", "This is currently being worked upon");
    private ErrorType(String type, String errorMsg) {
        this.type = type;
        this.errorMsg = errorMsg;
    }
    private String type;
    private String errorMsg;
    
    public static ErrorType getErrorMsgFor(String browserName) {
        if (IOSCapabilities.ipad().getBrowserName().equals(browserName)) {
            return IOS;
        }
        //rest of the code goes here
    }
}

```

So in the error msg section we would just do 
```
logger.log(Level.WARNING, ErrorType.getErrorMsgFor(((HasCapabilities) d).getCapabilities().getBrowserName()));
```

Would that work here ?